### PR TITLE
Replace Python tooling with Ruff & Pyright

### PR DESCRIPTION
**Pull request by disrupted**
Black, isort, flake8 → Ruff
MyPy → Pyright

### CODE REVIEW

File: templates/python-poetry/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+pytest = "^8.1.1"
+pytest-asyncio = "^0.23.6"
+pytest-mock = "^3.14.0"
+pytest-clarity = "^1.0.1"
+pytest-cov = "^5.0.0"
+
+[tool.ruff]
+# E203: whitespace before ':' -- Not PEP8 compliant, black won't correct it
+# E501: Line too long -- Clashes with `black`
+ignore = ["E203", "E501"]
```
Comment (sujuka99) [inline templates/python-poetry/pyproject.toml]:
```suggestion
[tool.ruff.lint]
ignore = [
  # Rules in conflict with `ruff-format` -- START
  "W191", # Checks for indentation that uses tabs. Spaces are preferred.
  # "E111", # Checks for indentation with a non-multiple of 4 spaces, add when out of nursery
  # "E114", # Checks for indentation of comments with a non-multiple of 4 spaces, add when out of nursery
  # "E117", # Checks for over-indented code, add when out of nursery
  "E501", # Line too long
  # Rules in conflict with `ruff-format` -- END
  "RUF012", # type class attrs with `ClassVar` -- Too strict/trigger-happy

```

File: templates/python-poetry/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# For a list of all possible rules visit https://beta.ruff.rs/docs/rules/
```
Comment (sujuka99) [inline templates/python-poetry/pyproject.toml]:
```suggestion
# For a list of all possible rules visit https://docs.astral.sh/ruff/rules/
```

File: templates/python-poetry/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+[tool.poetry.group.dev.dependencies]
+pre-commit = "^3.7.0"
+ruff = "^0.3.4"
+pyright = "^1.1.355"
+pytest = "^8.1.1"
+pytest-asyncio = "^0.23.6"
+pytest-mock = "^3.14.0"
+pytest-clarity = "^1.0.1"
+pytest-cov = "^5.0.0"
+
```
Comment (sujuka99) [inline templates/python-poetry/pyproject.toml:25]:
```suggestion

[tool.ruff]
output-format = "grouped"
show-fixes = true
target-version = "py310"

```

File: templates/python-poetry/pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+[tool.ruff]
+# E203: whitespace before ':' -- Not PEP8 compliant, black won't correct it
+# E501: Line too long -- Clashes with `black`
+ignore = ["E203", "E501"]
+# For a list of all possible rules visit https://beta.ruff.rs/docs/rules/
+# Pyflakes(F), pycodestyle(E,W), mccabe(C90), isort(I), pyupgrade(UP), ruff(RUF)
+select = ["F", "E", "W", "C90", "I", "UP", "RUF"]
```
Comment (sujuka99) [inline templates/python-poetry/pyproject.toml]:
```suggestion
[tool.ruff.lint]
select = [
  "F", # Pyflakes
  "E", # pycodestyle Errors
  "W", # pycodestyle Warnings
  "C90", # mccabe
  "I", # isort
  "UP", # pyupgrade
  "B", # flake8-bugbear
  "RUF", # Ruff-specific rules
```

**Review by sujuka99 — CHANGES_REQUESTED** [review]

File: templates/python/setup.cfg [hunk 5]
```diff
@@ -1,30 +0,0 @@
...
-disable_error_code = import
-# do not require explicit `return None` in functions with Optional return type
-warn_no_return = False
-# don't skip functions which are unannotated, such as tests
-check_untyped_defs = True
-
-[tool:pytest]
-testpaths =
-    tests
-addopts = --cov=package-name --cov-report term-missing
```
Comment (jkbe) [inline templates/python/setup.cfg:30]: Is it intentional to remove this without replacement? 

You can also add such options to pyproject.toml (I guess then they will only be respected when running `poetry run pytest` and not with bare `pytest` though?). I have this in another project:
```
[tool.pytest.ini_options]
markers = ["live: test to run a live server running on localhost:8080"]
addopts = "-m 'not live'"
```

Comment (disrupted) [reply templates/python/setup.cfg:30]: good point, I overlooked it. Will add those to pyproject.toml

**Review by sujuka99 — APPROVED** [review]

**Review by jkbe — APPROVED** [review]

**Pull request merged**